### PR TITLE
[FIXED JENKINS-42836] env expand the script path from SCM.

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-42836](https://issues.jenkins-ci.org/browse/JENKINS-42836)

cc @reviewbybees

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java [hunk 1]
```diff
@@ -181,4 +181,22 @@
...
+        sampleRepo.write("flow.groovy", "echo 'version one'");
+        sampleRepo.git("add", "flow.groovy");
+        sampleRepo.write("otherFlow.groovy", "echo 'version two'");
+        sampleRepo.git("add", "otherFlow.groovy");
+        sampleRepo.git("commit", "--all", "--message=commits");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(new GitSCM(Collections.singletonList(new UserRemoteConfig(sampleRepo.fileUrl(), null, null, null)),
+                Collections.singletonList(new BranchSpec("master")),
+                false, Collections.<SubmoduleConfig>emptyList(), null, null, Collections.<GitSCMExtension>emptyList()), "${SCRIPT_PATH}");
+        def.setLightweight(false); // TODO SCMFileSystem.of cannot pick up build parameters
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java]: No no, this is wrong, just copied-pasted from the previous test which dealt with SCM configuration. You _are_ expanding `scriptPath` even in lightweight mode. So either delete this line, or test it in each mode for full coverage.

Comment (abayer) [reply src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java]: whoops, good catch. Thanks!

**Comment by abayer** [comment]
Merge conflicts fixed.

**Comment by jglick** [comment]
[broken test](https://ci.jenkins.io/job/Plugins/job/workflow-cps-plugin/job/PR-197/10/testReport/org.jenkinsci.plugins.workflow.cps/CpsScmFlowDefinitionTest/retry/)

**Comment by abayer** [comment]
@jglick Yeah, that's [INFRA-1516](https://issues.jenkins-ci.org/browse/INFRA-1516)

**Comment by eyusupov** [comment]
@jglick looks like the CI failures were addressed.

**Pull request merged**